torch/storage.py

```diff
@@ -195,12 +194,35 @@ def complex_float(self):
         """Casts this storage to complex float type"""
         return self._to(torch.cfloat)
 
+    def is_pinned(self, device="cuda"):
+        """
+        Determine whether the CPU storage is already pinned on device.
+
+        :param device (str): String representation of device type to pin memory on.
+            If the device is not set, the default behavior is CUDA device.
+
+        :return: A boolean variable
+        """
+        import torch
+        if device == "cuda" or device == torch._C._get_privateuse1_backend_name():
+            return torch.tensor([], dtype=torch.uint8, device=self.device).set_(
```

Comment:
> I'd say we should probably just create the tensor all the time, with no conditional. That way `Tensor.is_pinned` takes care of absolutely everything, including support for both `str` and `torch.device` types for the `device` arg.
> 
> Also, use `torch.empty(0, ...)`, to avoid the little bit of overhead from allocating a new storage temporarily

Hi, @kurtamohler 
  I get what you mean now. The parameter  '''device'''  of  '''tenosr.is_pinnned''' will be used as dispatch key for ''' at::_ops::is_pinned '''. When calling ''' tensor.is_pinned("invalid_device_xxx")''', I think the user might want to return  ''' False ''' instead of  ''' ’is_pinned  is not implemented on the invalid_device_xxx'''.   In addition, this implementation of this method is closer to the previous implementation in C++.   Do you think we should change the default behavior?

Reply:
> If the given device isn't a valid device, I think it's better to throw an error, rather than returning `False` silently. If an invalid device is given, that indicates that the caller misspelled something or something else went wrong. Does that make sense?
> 
> Either way, `Tensor.is_pinned` and `UntypedStorage.is_pinned` should have the same behavior. So if we decide not to throw an error for invalid devices in `UntypedStorage.is_pinned`, then we should make that change in `Tensor.is_pinned` as well. And either way, we can still make `UntypedStorage.is_pinned` unconditionally wrap the storage in a temporary tensor and call `Tensor.is_pinned`

I can't agree with you more, and we really should keep the same behavior between `Tensor.is_pinned` and `UntypedStorage.is_pinned` . In the latest PR, this implementation has been updated as you suggested.

On the other hand, I found that what I wanted to do was already done, in the implementation of `Tensor.is_pinned`, by calling `Tensor.is_pinned_default` for those backends that do not register their own implementation of `Tensor.is_pinned`. 

I'm sincerely looking forward to your reviews, @kurtamohler.